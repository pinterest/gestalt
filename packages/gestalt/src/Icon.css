.icon {
  fill: currentColor;
  stroke-width: 0;
  vertical-align: middle;
}

.iconBlock {
  composes: block from "./Layout.css";
}
