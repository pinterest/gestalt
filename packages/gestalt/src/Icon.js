// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Icon.css';
import icons from './icons/index.js';
import colors from './Colors.css';

type IconProps = {
  accessibilityLabel: string,
  color?:
    | 'blue'
    | 'darkGray'
    | 'eggplant'
    | 'gray'
    | 'green'
    | 'lightGray'
    | 'maroon'
    | 'midnight'
    | 'navy'
    | 'olive'
    | 'orange'
    | 'orchid'
    | 'pine'
    | 'purple'
    | 'red'
    | 'watermelon'
    | 'white',
  inline?: boolean,
  size?: number | string,
};

type IconNoPath = {
  icon: $Keys<typeof icons>,
  dangerouslySetSvgPath?: null,
};

type PathNoIcon = {
  icon?: null,
  dangerouslySetSvgPath: { __path: string },
};

type Props = IconProps & (PathNoIcon | IconNoPath);

const IconNames = Object.keys(icons);

export default function Icon(props: Props) {
  const {
    accessibilityLabel,
    color = 'gray',
    dangerouslySetSvgPath,
    icon,
    inline,
    size = 16,
  } = props;

  const cs = classnames(styles.icon, colors[color], {
    [styles.iconBlock]: !inline,
  });

  const path =
    (icon && icons[icon]) ||
    /* eslint-disable-next-line no-underscore-dangle */
    (dangerouslySetSvgPath && dangerouslySetSvgPath.__path) ||
    undefined;

  const ariaHidden = accessibilityLabel === '' ? true : null;

  return (
    <svg
      className={cs}
      height={size}
      width={size}
      viewBox="0 0 24 24"
      aria-hidden={ariaHidden}
      aria-label={accessibilityLabel}
      role="img"
    >
      <path d={path} />
    </svg>
  );
}

Icon.icons = IconNames;

Icon.propTypes = {
  accessibilityLabel: PropTypes.string.isRequired,
  color: PropTypes.oneOf([
    'blue',
    'darkGray',
    'eggplant',
    'gray',
    'green',
    'lightGray',
    'maroon',
    'midnight',
    'navy',
    'olive',
    'orange',
    'orchid',
    'pine',
    'purple',
    'red',
    'watermelon',
    'white',
  ]),
  dangerouslySetSvgPath: PropTypes.shape({
    __path: PropTypes.string,
  }),
  icon: PropTypes.oneOf(IconNames),
  inline: PropTypes.bool,
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};
