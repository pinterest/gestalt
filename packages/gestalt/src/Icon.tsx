import classnames from 'classnames';
import styles from './Icon.css';
import icons from './icons/index';
import vrIcons from './icons-vr-theme/index';
import useInExperiment from './useInExperiment';

export type IconColor =
  | 'default'
  | 'disabled'
  | 'subtle'
  | 'success'
  | 'error'
  | 'warning'
  | 'info'
  | 'recommendation'
  | 'inverse'
  | 'shopping'
  | 'brandPrimary'
  | 'light'
  | 'dark';

type Props = {
  /**
   * Label for screen readers to announce Icon. This populates the `aria-label` attribute. If the label is hidden, use an empty string for the label (`accessibilityLabel=""`) to set `aria-hidden`.
   *
   * See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/icon#Accessibility) for details on proper usage.
   */
  accessibilityLabel: string;
  /**
   * The colors available to apply to Icon.
   *
   * See the [color variant](https://gestalt.pinterest.systems/web/icon#Colors) to learn more.
   */
  color?: IconColor;
  /**
   * Available for testing purposes, if needed. Consider [better queries](https://testing-library.com/docs/queries/about/#priority) before using this prop.
   */
  dataTestId?: string;
  /**
   * SVG icon from the Gestalt icon library to use within Icon.
   *
   * See the [icon library](https://gestalt.pinterest.systems/foundations/iconography/library) to explore available options.
   */
  icon?: keyof typeof icons;
  /**
   * Defines a new icon different from the built-in Gestalt icons.
   *
   * See the [custom icon](https://gestalt.pinterest.systems/web/icon#Custom-icon) variant to learn more.
   */
  dangerouslySetSvgPath?: {
    __path: string;
  };
  /**
   * Properly positions Icon relative to an inline element, such as Text using the inline property.
   */
  inline?: boolean;
  /**
   * Use a number for pixel sizes or a string for percentage based sizes.
   *
   * See the [size](https://gestalt.pinterest.systems/web/icon#Size) variant to learn more.
   */
  size?: number | string;
};

// @ts-expect-error - TS2322 - Type 'string[]' is not assignable to type 'readonly ("replace" | "search" | "link" | "text" | "dash" | "3D" | "3D-move" | "360" | "accessibility" | "ad" | "ad-group" | "add" | "add-circle" | "add-layout" | "add-pin" | "add-section" | ... 317 more ... | "wave")[]'.
const IconNames: ReadonlyArray<keyof typeof icons> = Object.keys(icons);

const flipOnRtlIconNames = [
  'ads-stats',
  'ads-overview',
  'arrow-back',
  'arrow-circle-forward',
  'arrow-end',
  'arrow-forward',
  'arrow-start',
  'arrow-up-right',
  'compose',
  'chevron-small-left',
  'chevron-small-right',
  'chevron-left-circle',
  'chevron-right-circle',
  'directional-arrow-left',
  'directional-arrow-right',
  'flipVertical',
  'hand-pointing',
  'link',
  'mute',
  'reorder-images',
  'send',
  'sound',
  'speech',
  'speech-ellipsis',
  'switch-account',
  'text-size',
  'visit',
];

/**
 * [Icons](https://gestalt.pinterest.systems/web/icon) are the symbolic representation of an action or information, providing visual context and improving usability.
 *
 * See the [Iconography and SVG guidelines](https://gestalt.pinterest.systems/foundations/iconography/library) to explore the full icon library.
 *
 * ![Icon light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Icon-list.spec.ts-snapshots/Icon-list-chromium-darwin.png)
 * ![Icon dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Icon-list-dark.spec.ts-snapshots/Icon-list-dark-chromium-darwin.png)
 *
 */

function Icon({
  accessibilityLabel,
  color = 'subtle',
  dangerouslySetSvgPath,
  dataTestId,
  icon,
  inline = false,
  size = 16,
}: Props) {
  const cs = classnames(
    // @ts-expect-error - TS2345 - Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
    flipOnRtlIconNames.includes(icon) && styles.rtlSupport,
    styles[color],
    styles.icon,
    { [styles.iconBlock]: !inline },
  );
  const isInExperiment = useInExperiment({
    webExperimentName: 'web_gestalt_visualRefresh',
    mwebExperimentName: 'web_gestalt_visualRefresh',
  });
  const path =
    (icon && (isInExperiment ? vrIcons : icons)[icon]) ||
    /* eslint-disable-next-line no-underscore-dangle */
    (dangerouslySetSvgPath && dangerouslySetSvgPath.__path) ||
    undefined;

  const ariaHidden = accessibilityLabel === '' ? true : null;

  return (
    // @ts-expect-error - TS2322 - Type '{ children: Element; "aria-hidden": true | null; "aria-label": string; className: string; height: string | number; role: "img"; viewBox: string; width: string | number; }' is not assignable to type 'SVGProps<SVGSVGElement>'.
    <svg
      aria-hidden={ariaHidden}
      aria-label={accessibilityLabel}
      className={cs}
      data-test-id={dataTestId}
      height={size}
      role="img"
      viewBox="0 0 24 24"
      width={size}
    >
      <path d={path} />
    </svg>
  );
}

Icon.icons = IconNames;

Icon.displayName = 'Icon';

export default Icon;
