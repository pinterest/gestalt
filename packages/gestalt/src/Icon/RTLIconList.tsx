import icons from '../icons/index';
import compactIconsVR from '../icons-vr-theme/compact/index';

type IconName = keyof typeof icons | keyof typeof compactIconsVR;

const swapOnRtlIconNames: ReadonlyArray<IconName> = Object.freeze(['list-numbered']);

const flipOnRtlIconNames: ReadonlyArray<IconName | undefined> = Object.freeze([
  'ads-stats',
  'ads-overview',
  'arrow-back',
  'arrow-circle-forward',
  'arrow-end',
  'arrow-forward',
  'arrow-start',
  'arrow-up-right',
  'compose',
  'chevron-left-circle',
  'chevron-right-circle',
  'directional-arrow-left',
  'directional-arrow-right',
  'flip-vertical',
  'hand-pointing',
  'link',
  'mute',
  'reorder-images',
  'send',
  'sound',
  'speech',
  'speech-ellipsis',
  'switch-account',
  'text-size',
  'visit',
]);

export { flipOnRtlIconNames, swapOnRtlIconNames };
