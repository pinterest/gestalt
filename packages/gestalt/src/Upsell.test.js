// @flow strict
import React from 'react';
import { create } from 'react-test-renderer';
import Icon from './Icon.js';
import Upsell from './Upsell.js';

describe('<Upsell />', () => {
  test('Basic Upsell', () => {
    const tree = create(<Upsell message="Insert a clever upsell message here" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title', () => {
    const tree = create(
      <Upsell message="Insert a clever upsell message here" title="A Title" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction with href', () => {
    const tree = create(
      <Upsell
        message="Insert a clever upsell message here"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest' }}
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction without href', () => {
    const tree = create(
      <Upsell
        message="Insert a clever upsell message here"
        primaryAction={{ label: 'Visit Pinterest' }}
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + secondaryAction', () => {
    const tree = create(
      <Upsell
        message="Insert a clever upsell message here"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest' }}
        secondaryAction={{ href: 'pinterest.com/help', label: 'Learn more' }}
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + dismissButton', () => {
    const tree = create(
      <Upsell
        message="Insert a clever upsell message here"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest' }}
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: () => {},
        }}
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + dismissButton + image', () => {
    const tree = create(
      <Upsell
        message="Insert a clever upsell message here"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest' }}
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: () => {},
        }}
        title="A Title"
        imageData={{
          component: <Icon icon="pinterest" accessibilityLabel="Pin" color="darkGray" size={32} />,
        }}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
