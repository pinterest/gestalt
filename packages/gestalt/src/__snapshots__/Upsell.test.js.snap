// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upsell /> Basic Upsell 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    />
  </div>
</div>
`;

exports[`<Upsell /> message + title + dismissButton + image + form 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box flexGrow itemsCenter justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box flexNone marginBottom4 smMarginBottom0"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="Mask rounding0 willChangeTransform"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        >
          <svg
            aria-hidden={null}
            aria-label="Pin"
            className="icon darkGray iconBlock"
            height={32}
            role="img"
            viewBox="0 0 24 24"
            width={32}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
      <div
        className="box flexGrow itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart6 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
        <div
          className="box flexGrow justifyEnd smDisplayFlex smMarginEnd4 smPaddingY3"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <form
            onSubmit={[Function]}
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="Flex rowGap2 flexWrap justifyEnd xsDirectionRow"
            >
              <div
                className="FlexItem flexGrow"
              >
                <div
                  className="box marginBottom0 smMarginBottom2"
                >
                  <span>
                    <input
                      aria-describedby={null}
                      aria-invalid="false"
                      className="textField base enabled normal medium"
                      disabled={false}
                      id="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Name"
                      type="text"
                    />
                  </span>
                </div>
              </div>
              <div
                className="FlexItem"
              >
                <button
                  aria-label="Submit button"
                  className="button hideOutline tapTransition md red enabled inline accessibilityOutline"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="submit"
                >
                  <div
                    className="Text fontSize3 white alignCenter fontWeightBold"
                  >
                    Submit
                  </div>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss banner"
      className="button tapTransition enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="pog transparent"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      >
        <svg
          aria-hidden={true}
          aria-label=""
          className="icon darkGray iconBlock"
          height={16}
          role="img"
          viewBox="0 0 24 24"
          width={16}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`<Upsell /> message + title + primaryAction + dismissButton + image 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box flexNone marginBottom4 smMarginBottom0"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <div
          className="Mask rounding0 willChangeTransform"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        >
          <svg
            aria-hidden={null}
            aria-label="Pin"
            className="icon darkGray iconBlock"
            height={32}
            role="img"
            viewBox="0 0 24 24"
            width={32}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart6 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <a
          className="link hideOutline tapTransition block pill accessibilityOutline buttonLink button lg red"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Text fontSize3 white alignCenter fontWeightBold"
          >
            Visit Pinterest
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss banner"
      className="button tapTransition enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="pog transparent"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      >
        <svg
          aria-hidden={true}
          aria-label=""
          className="icon darkGray iconBlock"
          height={16}
          role="img"
          viewBox="0 0 24 24"
          width={16}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`<Upsell /> message + title + primaryAction + dismissButton 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <a
          className="link hideOutline tapTransition block pill accessibilityOutline buttonLink button lg red"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Text fontSize3 white alignCenter fontWeightBold"
          >
            Visit Pinterest
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss banner"
      className="button tapTransition enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="pog transparent"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      >
        <svg
          aria-hidden={true}
          aria-label=""
          className="icon darkGray iconBlock"
          height={16}
          role="img"
          viewBox="0 0 24 24"
          width={16}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`<Upsell /> message + title + primaryAction + secondaryAction 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <a
          className="link hideOutline tapTransition block pill accessibilityOutline buttonLink button lg gray"
          href="pinterest.com/help"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Text fontSize3 darkGray alignCenter fontWeightBold"
          >
            Learn more
          </div>
        </a>
      </div>
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <a
          className="link hideOutline tapTransition block pill accessibilityOutline buttonLink button lg red"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Text fontSize3 white alignCenter fontWeightBold"
          >
            Visit Pinterest
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Upsell /> message + title + primaryAction with href 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <a
          className="link hideOutline tapTransition block pill accessibilityOutline buttonLink button lg red"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onFocus={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Text fontSize3 white alignCenter fontWeightBold"
          >
            Visit Pinterest
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Upsell /> message + title + primaryAction without href 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box itemsCenter justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock"
      >
        <button
          className="button hideOutline tapTransition lg red enabled block accessibilityOutline"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="Text fontSize3 white alignCenter fontWeightBold"
          >
            Visit Pinterest
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Upsell /> message + title 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 shadow smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box itemsCenter justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex"
    >
      <div
        className="box itemsCenter marginBottomAuto marginEnd0 marginStart0 marginTopAuto smDisplayBlock smMarginEnd6 smMarginStart0 xsDirectionColumn xsDisplayFlex"
      >
        <div
          className="box"
          style={
            Object {
              "maxWidth": 648,
            }
          }
        >
          <div
            className="box marginBottom2"
          >
            <h3
              className="Heading fontSize1 darkGray alignLeft breakWord"
            >
              A Title
            </h3>
          </div>
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal"
          >
            Insert a clever upsell message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    />
  </div>
</div>
`;
